lib/ddtrace/profiling/profiler.rb

```diff
@@ -0,0 +1,28 @@
+module Datadog
+  # Generates profiles and transmits them to Datadog
+  class Profiler
+    attr_reader \
+      :collectors,
+      :scheduler
+
+    def initialize(collectors, scheduler)
```

Comment:
That's fine, though I imagine you want something a little more "high level" for end users. They are not going to pass collectors and schedulers :)

Reply:
Yeah, I don't think end users will be initializing a `Profiler` directly. The logic necessary to actually construct a complete `Profiler` is quite complex, as all the individual parts need to be orchestrated together, essentially in a factory pattern. `Datadog.configure` does this, so users can simply access the profiler via `Datadog.profiler`.

This is largely because we have a `Settings` object that has lots of different configuration options, and I don't want to couple the `Profiler` itself with the configuration settings object. Originally I tried to pass through all the individual settings from the configuration object as options in via `def initialize(options = {})` but this looked too complicated and duplicative of the configuration settings object.